gumby/modules/tribler_module.py

```diff
@@ -0,0 +1,42 @@
+from os import environ, path, getpid, makedirs, symlink
+
+from twisted.internet.threads import deferToThread
+
+from gumby.sync import experiment_callback
+from gumby.modules.experiment_module import static_module
+from gumby.modules.base_dispersy_module import BaseDispersyModule
+
+
+@static_module
+class TriblerModule(BaseDispersyModule):
+    def __init__(self, experiment):
+        super(TriblerModule, self).__init__(experiment)
+        # in the case of Tribler dispersy we cannot control the launch of the discovery community. So don't try it.
+        self.custom_community_loader.del_launcher("DiscoveryCommunity")
+
+    @experiment_callback
+    def start_session(self):
+        self._logger.error("Starting Tribler Session")
+
+        if self.custom_community_loader:
+            self.session.lm.community_loader = self.custom_community_loader
+
+        def on_tribler_started(_):
+            self._logger.error("Tribler Session started")
+            self.dispersy = self.session.lm.dispersy
+
+        return self.session.start().addCallback(on_tribler_started)
+
+    @experiment_callback
+    def stop_session(self):
+        deferToThread(self.session.shutdown)
+
+    def setup_session_config(self):
+        # symlink the bootstrap file so we are only connecting to our own trackers
```

Comment:
We don't always want to connect to only our trackers (i.e. when we run Tribler idle) so we should make this optional.